### PR TITLE
Include .d.ts in srcs in the transitive_declarations passed from node…

### PR DESCRIPTION
**Pull request by alexeagle**
…_module_library to ts_library

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: internal/npm_install/node_module_library.bzl [hunk 1]
```diff
@@ -32,8 +32,8 @@ def _node_module_library_impl(ctx):
     # declarations are a subset of sources that are decleration files
     declarations = depset([f for f in ctx.files.srcs if f.path.endswith(".d.ts")])
 
-    # transitive_declarations are all direct & transitive decleration files
-    transitive_declarations = depset()
+    # transitive_declarations are all .d.ts files in srcs plus those in direct & transitive dependencies
+    transitive_declarations = depset(transitive = [declarations])
     for src in ctx.attr.srcs:
```
Comment (alexeagle) [inline internal/npm_install/node_module_library.bzl]: does it really make sense that node_module_library might have non-src entries in srcs? Seems to me we should only walk deps[] to collect the transitive

**Comment by alexeagle** [comment]
should fix rules_typescript#381

File: internal/npm_install/node_module_library.bzl (hunk 1 above)
Comment (alexeagle) [reply internal/npm_install/node_module_library.bzl]: per slack, removed the walk over srcs

Comment (gregmagolan) [reply internal/npm_install/node_module_library.bzl]: yup. srcs should only be files or filegroup so they have no "typescript" provider to walk over

**Comment by alexeagle** [comment]
Now simply using the upstream rules_typescript instead of a fork, means no followup cleanup change later

**Pull request merged**